crates/fuel-core/src/service/genesis/exporter.rs

```diff
@@ -57,12 +57,13 @@ where
         writer: Fun,
         group_size: usize,
     ) -> Self {
+        let (_, receiver) = watch::channel(State::Started);
         Self {
             db,
             prev_chain_config,
             writer,
             group_size,
-            task_manager: TaskManager::new(CancellationToken::new()),
+            task_manager: TaskManager::new(receiver.into()),
```

Comment:
Do we even use it in the case of exporter? It would be nice to avoid passing unused parameters. Or maybe it makes sense to process export cancelation as well=)

Reply:
It probably would be more user friendly if the process could be stopped. 

Do something like write into a tempdir in the same fs (so that moving/renaming is cheap), if stopped remove the temp dir. If not stopped, at the end just rename the dir to the location originally indented for the snapshot.

I could implement that in a follow up PR if you agree.

Reply:
I think we don't need to bother with temporary storage and resumability=) We can just shut the process and next time start from scratch.

Reply:
If it is hard to integrate in this PR, you can leave a TODO comment with the link to the issue and do in follow up PR

Reply:
I'll do it in this PR. 

> I think we don't need to bother with temporary storage and resumability=) 

I'd just implement the 'rename-when-finished-delete-temp-if-cancelled` if that's ok. Seems better to leave the filesystem clean if the snapshot generation was cancelled. No way of generating a broken snapshot then (barring ungraceful shutdown ofc but even then it would be some temp file that would be left).